test/framework/cluster_proxy.go

```diff
@@ -66,6 +66,9 @@ type ClusterProxy interface {
 	// Apply to apply YAML to the Kubernetes cluster, `kubectl apply`.
 	Apply(context.Context, []byte) error
 
+	// Apply to apply YAML to the Kubernetes cluster, `kubectl apply` with extra args.
+	ApplyWithArgs(ctx context.Context, resources []byte, args ...string) error
```

Comment:
Could we change the Apply signature instead here?

Reply:
There is already an implementation for [ApplyWithArgs](https://github.com/kubernetes-sigs/cluster-api/blob/34e62089fa3378fc3391d4f2e7276b581c790d2e/test/framework/cluster_proxy.go#L191) in `cluster_proxy`. Do you want me to change `Apply` to do what `ApplyWithArgs` is doing?

Reply:
@fabriziopandini wdyt? It makes sense to me to have only one, but I'm ok either way if you all think this is better

Reply:
I'm +1 for having only one method given that we can make breaking changes